Drop support for legacy CA-based auth (#369)
GalaChain/sdk: docs/authorization.md

# Authorization and authentication

GalaChain uses two layers of authorization and authentication to ensure that only authorized users can access the system.
First level, exposed to the client, is based on secp256k1 signatures and private/public key authorization for Ethereum signing scheme, and eddsa signatures for TON signing scheme.

> Second level uses native Hyperledger Fabric CA users and organizations MSPs.

Reviewer: ```suggestion
The second level uses native Hyperledger Fabric CA users and organizations MSPs.
```

> Signature based authorization user secp256k1 signatures to verify the identity of the end user.

Reviewer: ```suggestion
Signature-based authorization uses secp256k1 signatures to verify the identity of the end user.
```

> By default, it uses the same algorithm as Ethereum (keccak256 + secp256k1), but also TON (The Open Network) signing scheme is supported.

Reviewer: ```suggestion
By default, it uses the same algorithm as Ethereum (keccak256 + secp256k1), but the TON (The Open Network) signing scheme is also supported.
```

> All payloads that should be authorized with TON signing scheme must have the `signing` field set to `TON`.

Reviewer: ```suggestion
All payloads that should be authorized with the TON signing scheme must have the `signing` field set to `TON`.
```

For the `PublicKeyContract` chaincode, the `CURATOR_ORG_MSP` environment variable is used as the organization that is allowed to register users (default value is `CuratorOrg`).
It is recommended to use the same variable for curator-level access to the chaincode methods.

## Role Based Access Control (RBAC)

GalaChain SDK v2 introduced a Role Based Access Control (RBAC) system that will allow for more fine-grained control over who can access what resources.

The `allowedOrgs` property is deprecated and will be eventually removed from the chaincode definition.
Instead, you should use the `allowedRoles` property to specify which **roles** can access the method.

The `allowedRoles` property is an array of strings that represent the roles that are allowed to access the method.
The roles are assigned to the `UserProfile` object in the chain data.
By default, the `EVALUATE` and `SUBMIT` roles are assigned to the user when they are registered.

> You can assign additional roles to the user using the `PublicKeyContract:UpdateUserRoles` method (requires `CURATOR` role or being called by CA user from a curator organization).

Reviewer: ```suggestion
You can assign additional roles to the user using the `PublicKeyContract:UpdateUserRoles` method. This method requires that the calling user either has the `CURATOR` role or is a CA user from a curator organization.
```

> There are some predefined roles (`EVALUATE`, `SUBMIT`, `CURATOR`), you can also define custom roles for more granular access control.

Reviewer: ```suggestion
There are some predefined roles (`EVALUATE`, `SUBMIT`, `CURATOR`). You can also define custom roles for more granular access control.
```
